File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.java
import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        int arr[][] = new int[n][2];

        for(int i = 0; i < arr.length; i++){
            st = new StringTokenizer(br.readLine(), " ");
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr,(arr1,arr2) -> {
            if(arr1[1] - arr2[1] == 0) return arr1[0] - arr2[0];
            else return arr1[1] - arr2[1];
        });

        for(int i = 0; i < arr.length; i++){
            sb.append(arr[i][0]).append(" ").append(arr[i][1]).append("\n");
        }
        bw.write(sb.toString());
        bw.flush();
    }
}

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sb = StringBuilder()

    val n = br.readLine().toInt()
    var arr = ArrayList<Pair<Int, Int>>()

    for (i in 0 until n) {
        val input = br.readLine().split(" ")
        val x = input[0].toInt()
        val y = input[1].toInt()

        arr.add(Pair(x, y))
    }
    arr.sortWith { o1, o2 ->
        if (o1.second == o2.second) {
            o1.first - o2.first
        } else {
            o1.second - o2.second
        }
    }

    for ((x, y) in arr) {
        sb.append(x).append(" ").append(y).append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}